backbone/src/main/java/org/researchstack/backbone/task/OrderedTask.java

```diff
@@ -67,6 +67,10 @@ public Step getStepAfterStep(Step step, TaskResult result) {
             return steps.get(0);
         }
 
+        if (step.isCompletion()) {
```

Comment:
It would ultimately read much better: 

`if (step.isFinalStepOfTheFlow()) return`